lidar_euclidean_cluster_detect/nodes/lidar_euclidean_cluster_detect/lidar_euclidean_cluster_detect.cpp

```diff
@@ -224,33 +224,58 @@ void publishCloudClusters(const ros::Publisher *in_publisher, const autoware_msg
     autoware_msgs::CloudClusterArray clusters_transformed;
     clusters_transformed.header = in_header;
     clusters_transformed.header.frame_id = in_target_frame;
-    for (auto i = in_clusters.clusters.begin(); i != in_clusters.clusters.end(); i++)
+    for (const auto & cluster : in_clusters.clusters)
     {
       autoware_msgs::CloudCluster cluster_transformed;
       cluster_transformed.header = in_header;
       try
       {
         _transform_listener->lookupTransform(in_target_frame, _velodyne_header.frame_id, ros::Time(),
                                              *_transform);
-        pcl_ros::transformPointCloud(in_target_frame, *_transform, i->cloud, cluster_transformed.cloud);
-        _transform_listener->transformPoint(in_target_frame, ros::Time(), i->min_point, in_header.frame_id,
+        pcl_ros::transformPointCloud(in_target_frame, *_transform, cluster.cloud, cluster_transformed.cloud);
+        _transform_listener->transformPoint(in_target_frame, ros::Time(), cluster.min_point, in_header.frame_id,
                                             cluster_transformed.min_point);
-        _transform_listener->transformPoint(in_target_frame, ros::Time(), i->max_point, in_header.frame_id,
+        _transform_listener->transformPoint(in_target_frame, ros::Time(), cluster.max_point, in_header.frame_id,
                                             cluster_transformed.max_point);
-        _transform_listener->transformPoint(in_target_frame, ros::Time(), i->avg_point, in_header.frame_id,
+        _transform_listener->transformPoint(in_target_frame, ros::Time(), cluster.avg_point, in_header.frame_id,
                                             cluster_transformed.avg_point);
-        _transform_listener->transformPoint(in_target_frame, ros::Time(), i->centroid_point, in_header.frame_id,
+        _transform_listener->transformPoint(in_target_frame, ros::Time(), cluster.centroid_point, in_header.frame_id,
                                             cluster_transformed.centroid_point);
 
-        cluster_transformed.dimensions = i->dimensions;
-        cluster_transformed.eigen_values = i->eigen_values;
-        cluster_transformed.eigen_vectors = i->eigen_vectors;
+        // Convex_hull
+        geometry_msgs::PolygonStamped polygon_tfed;
+        for (geometry_msgs::Point32 point : cluster.convex_hull.polygon.points) {
```

Comment:
This could also be `const auto & point` to avoid a copy for each point.

Reply:
I used a classical for loop to solve your next request more easily.